app/models/user.rb

```diff
@@ -46,16 +47,16 @@ def remember
     update_attributes remember_digest: User.digest(remember_token)
   end
 
+  def profile
+    super || build_profile
```

Comment:
thế này không ổn nhé. a vẫn muốn logic là khi hiển thị mà user không có profile thì profile của nó phải là nil.

Reply:
method accepts_nested_attributes_for(*attr_names) 
với params là tên attributes 
hỗ trợ các option: 
:allow_destroy, Nếu true, destroys bất kỳ members nào từ các attributes
:reject_if, Cho phép chỉ định một Proc hoặc một Symbol, nó sẽ gọi với 1 hash của attributes. Nếu false thì sẽ không build đc object đó
:limit, Cho phép chỉ định số lượng tối đa của record associated có thể processed trong nested attributes
:update_only, mặc định là tắt và nested attributes sẽ update record đã tồn tại nếu include :id, nếu không có thì tạo record mới và sử dụng
còn nếu dùng :update_only thì nested attributes sẽ luôn update record bất kể :id có tồn tại hay không

Reply:
giải thích lại cho a các options của `accepts_nested_attributes` nhé

Reply:
:allow_destroy, mặc định là tắt
xóa 1 record thông qua attributes hash khi thêm _destroy vào attributes hash member.profile_attributes = { id: '2', _destroy: '1' }
:reject_if 
thiết lập một: reject_if proc hoặc symbol để bỏ bất kỳ bản ghi thông qua attribute hash nếu không thỏa mãn yêu cầu
accepts_nested_attributes_for :avatar, reject_if: proc { |attributes| attributes['name'].blank? }
nếu attributes 'name' trống thì không lưu record này
:limit, xác định tối đa record có thể xử lí trong nested attributes 
:update_only, mặc định là tắt và nested attributes sẽ update record đã tồn tại nếu include :id, nếu không có thì tạo record mới và sử dụng
còn nếu dùng :update_only thì nested attributes sẽ luôn update record bất kể :id có tồn tại hay không

Reply:
update_only hoạt động cho những association one to one ạ

:reject_if, cách dùng: chỉ định 1 proc hoặc 1 symbol sau đó gọi 1 method
dùng để kiểm tra xem bản ghi có được built cho attribute hash hay không.